### ISSUE TITLE
Background Sync Enhancements

### ISSUE THREAD

**Issue by TheScientist** [nextcloud/windows-universal#187]
I have a partially working background task version of the file sync:
https://github.com/TheScientist/windows-universal/tree/%2310_background
I'll track the todos here.

Version 1.2.0:
- [ ] SyncTask gets cancelled regularely with errorCode 0x80010008
- [ ] Refactoring of duplicate ClientService
- [ ] EventListener for better PushNotifications (notification per file)
- [ ] KeepAsIs ConflictSolution
- [ ] Handle Progress and Cancel in Background Task
- [ ] Settings for timetrigger + notifications
- [ ] Use built in sqlite (see #117)

Future versions
- [ ] Better Sync Algorithm (per Sync framework)to be able to delta sync

**Comment by DecaTec** [comment]
The background task will be outproc, right?

**Comment by TheScientist** [comment]
Yes, we need to move some of the classes into a class library, to make them usable by the bg task.  The branch has a partially working version. Sync starts but gets canceled sometimes.

Von: DecaTec
Gesendet: Sonntag, 11. Juni 2017 15:12
An: nextcloud/windows-universal
Cc: Stefan; Author
Betreff: Re: [nextcloud/windows-universal] Background Sync Enhancements (#187)

The background task will be outproc, right?

**Comment by TheScientist** [comment]
Looking at the code from the new [owncloud client](https://github.com/owncloud/OwncloudUniversal), I found 2 interesting things:

- they query a windows search index to look for locally changed files
- they compare local files by size to check for changes (we use modified time)

**Comment by SunboX** [comment]
Oh, nice, a universal owncloud client. Thats a great finding for reference! Let's see if can adapt some things!
